app/controllers/hq/sponsors_controller.rb

```diff
@@ -47,6 +48,12 @@ def load_sponsor
     @sponsor= Sponsor.find(params[:id])
   end
 
+  def load_active_and_inactive_sponsorships
+    sponsorships = Sponsorship.where("sponsor_id = ?", @sponsor.id).eager_load(:orphan, :sponsor).all
```

Comment:
eager loading the `:sponsor` too was necessary because the `sponsorship.sponsor_id` delegated to `sponsor.id`. But since that was fixed the `:sponsor` is not needed here.
Good find! Thanks :+1: